contracts/libraries/VaultLifecycle.sol

```diff
@@ -716,6 +720,94 @@ library VaultLifecycle {
         );
     }
 
+    /**
+     * @notice Allocates the vault's minted options to the OptionsPurchaseQueue contract
+     * @dev Skipped if the optionsPurchaseQueue doesn't exist
+     * @param optionsPurchaseQueue is the OptionsPurchaseQueue contract
+     * @param option is the minted option
+     * @param optionsAmount is the amount of options minted
+     * @param optionAllocation is the maximum % of options to allocate towards the purchase queue (will only allocate
+     *  up to the amount that is on the queue)
+     * @return allocatedOptions is the amount of options that ended up getting allocated to the OptionsPurchaseQueue
+     */
+    function allocateOptions(
+        address optionsPurchaseQueue,
+        address option,
+        uint256 optionsAmount,
+        uint256 optionAllocation
+    ) external returns (uint256 allocatedOptions) {
+        // Skip if optionsPurchaseQueue is address(0)
+        if (optionsPurchaseQueue != address(0)) {
+            allocatedOptions = optionsAmount.mul(optionAllocation).div(
+                100 * Vault.OPTION_ALLOCATION_MULTIPLIER
+            );
+            allocatedOptions = IOptionsPurchaseQueue(optionsPurchaseQueue)
+                .getOptionsAllocation(address(this), allocatedOptions);
+
+            if (allocatedOptions != 0) {
```

Comment:
Do we need to have this check? We have a similar check here: https://github.com/ribbon-finance/ribbon-v2/pull/241/files#diff-6d059eef33889a02d96f146813863a0b1a958de6e3730d8ef14ecdc88f4db0e3R207-R216